cmd/completion.go

```diff
@@ -0,0 +1,71 @@
+package cmd
+
+import (
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+var completionCmd = &cobra.Command{
+	Use:   "completion [bash|zsh|fish|powershell]",
+	Short: "Generate completion script",
+	Long: `To load completions:
+
+Bash:
+
+  $ source <(deck completion bash)
+
+  # To load completions for each session, execute once:
+  # Linux:
+  $ deck completion bash > /etc/bash_completion.d/deck
+  # macOS:
+  $ deck completion bash > /usr/local/etc/bash_completion.d/deck
+
+Zsh:
+
+  # If shell completion is not already enabled in your environment,
+  # you will need to enable it.  You can execute the following once:
+
+  $ echo "autoload -U compinit; compinit" >> ~/.zshrc
+
+  # To load completions for each session, execute once:
+  $ deck completion zsh > "${fpath[1]}/_yourprogram"
+
+  # You will need to start a new shell for this setup to take effect.
+
+fish:
+
+  $ deck completion fish | source
+
+  # To load completions for each session, execute once:
+  $ deck completion fish > ~/.config/fish/completions/deck.fish
+
+PowerShell:
+
+  PS> deck completion powershell | Out-String | Invoke-Expression
+
+  # To load completions for every new session, run:
+  PS> deck completion powershell > deck.ps1
+  # and source this file from your PowerShell profile.
+`,
+	DisableFlagsInUseLine: true,
+	ValidArgs:             []string{"bash", "zsh", "fish", "powershell"},
```

Comment:
optional:

The list of valid choices is defined twice: once in `ValidArgs`, then in the `switch`.

Suggestion to avoid duplication: remove `ValidArgs`, replace `cobra.ExactValidArgs` with `cobra.ExactArgs` and add a `default` case to the `switch`.

Reply:
Good call. Incorporated this change!